iroh-net/src/magicsock/node_map/endpoint.rs

```diff
@@ -286,6 +294,28 @@ impl Endpoint {
         }
     }
 
+    /// Returns a [`oneshot::Receiver`] that will be alerted once we have a valid `best_addr`
+    ///
+    /// If we already have a valid `best_addr` for this endpoint, the [`oneshot::Receiver`]
+    /// will recv immediately.
+    ///
+    /// It is possible we will never have a viable `best_addr`, in which case
+    /// no alert will ever be issued.
+    pub fn notify_has_best_addr(&mut self) -> oneshot::Receiver<()> {
+        self.assign_best_addr_from_candidates_if_empty();
```

Comment:
Do you really want to call this from here?  I'm kind of -1 because it is logic that influences how the magicsock behaves but is called from a place that's not in the normal flow.  The magicsock will already call this from somewhere else when it thinks this is needed.

Reply:
good point. if we go back to that flow, I'll remove it.